### PR TITLE
Improve help for conan remove command

### LINKED ISSUE #5495: Improve help for conan remove <ref_or_pattern> -p
Related to #1467, #1314

`conan remove <ref_or_pattern> -p` command is allowed and displayed in the docs but the help
does not state this clear

```
$ conan remove -h
usage: conan remove [-h] [-b [BUILDS [BUILDS ...]]] [-f] [-l] [-o]
                    [-p [PACKAGES [PACKAGES ...]]] [-q QUERY] [-r REMOTE] [-s]
                    [-t]
                    [pattern_or_reference]

Removes packages or binaries matching pattern from local cache or remote.

It can also be used to remove temporary source or build folders in the
local conan cache. If no remote is specified, the removal will be done
by default in the local conan cache.

positional arguments:
  pattern_or_reference  Pattern or package recipe reference, e.g., 'boost/*',
                        'MyPackage/1.2@user/channel'

optional arguments:
  -h, --help            show this help message and exit
  -b [BUILDS [BUILDS ...]], --builds [BUILDS [BUILDS ...]]
                        By default, remove all the build folders or select
                        one, specifying the package ID
  -f, --force           Remove without requesting a confirmation
  -l, --locks           Remove locks
  -o, --outdated        Remove only outdated from recipe packages. This flag
                        can only be used with a reference
  -p [PACKAGES [PACKAGES ...]], --packages [PACKAGES [PACKAGES ...]]
                        Select package to remove specifying the package ID
```

Conan 1.17

### PR DESCRIPTION
**Pull request by sdmg15**
Changelog: (Feature | Fix | Bugfix): 
Fix  #5495 
Docs: https://github.com/conan-io/docs/pull/XXXX

- [x] Refer to the issue that supports this Pull Request.
- [x] If the issue has missing info, explain the purpose/use case/pain/need that covers this Pull Request.
- [x] I've read the [Contributing guide](https://github.com/conan-io/conan/blob/develop/.github/CONTRIBUTING.md).
- [ ] I've followed the PEP8 style guides for Python code.
- [x] I've opened another PR in the Conan docs repo to the ``develop`` branch, documenting this one. 

<sup>**Note:** By default this PR will skip the slower tests and will use a limited set of python versions. Check [here](https://github.com/conan-io/conan/blob/develop/.github/PR_INCREASE_TESTING.md) how to increase the testing level by writing some tags in the current PR body text.</sup>

### CODE REVIEW

File: conans/client/command.py [hunk 1]
```diff
@@ -994,7 +994,7 @@ def remove(self, *args):
                             help="Remove only outdated from recipe packages. "
                                  "This flag can only be used with a reference")
         parser.add_argument('-p', '--packages', nargs="*", action=Extender,
-                            help="Select package to remove specifying the package ID")
+                            help="Remove all packages of the specified reference")
```
Comment (memsharded) [inline conans/client/command.py]: This is good, but the previous one is still valid. ``--packages=xxxxx`` will remove that package with the xxxxx ID, and ``--packages`` will remove all packages.

Comment (sdmg15) [reply conans/client/command.py]: Okay then I think "Remove all packages of the specified reference if no specific package ID is provided"  describe the both ?

Comment (memsharded) [reply conans/client/command.py]: Yes, good

**Comment by memsharded** [comment]
Thanks!

**Pull request merged**